lib/compress/zstd_compress.c

```diff
@@ -1121,18 +1169,18 @@ static void ZSTD_overrideCParams(
 }
 
 ZSTD_compressionParameters ZSTD_getCParamsFromCCtxParams(
-        const ZSTD_CCtx_params* CCtxParams, U64 srcSizeHint, size_t dictSize)
+        const ZSTD_CCtx_params* CCtxParams, U64 srcSizeHint, size_t dictSize, ZSTD_cParamMode_e mode)
```

Comment:
idea for discussion :
does it make sense to consider having `ZSTD_cParamMode_e mode` as a member of `CCtxParams`, rather than a separate parameter ?